### PR TITLE
DAOS-11390 test: Add NLT/ftest support for control_metadata

### PR DESCRIPTION
**Pull request by mjmac**
Enable use of an external directory for control metadata by
default. Eventually we'll want to add support for testing with
a dedicated control metadata device with hw tests, as well
as fallback tests for the old SCM-based metadata mode.

Features: control
Required-githooks: true

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/server_utils.py [hunk 1]
```diff
@@ -230,6 +230,20 @@ def clean_files(self, verbose=True):
...
+            # Remove the control metadata directory
+            self.log.info("Cleaning up the %s directory.", str(path))
+            clean_commands.append("sudo rm -fr {}".format(path))
+
+        if self.manager.job.yaml.control_metadata.device:
+            device = self.manager.job.yaml.control_metadata.device
+            # Clean up the control metadata device
+            self.log.info("Cleaning up the %s device.", str(device))
+            clean_commands.append("sudo umount {}".format(device))
+            clean_commands.append("sudo wipefs -a {}".format(device))
```
Comment (kjacque) [inline src/tests/ftest/util/server_utils.py:245]: You'll probably want to reverse the order of these, since the path is the mount point for the device.

Comment (mjmac) [reply src/tests/ftest/util/server_utils.py:245]: Oh, I typoed the first one. I was thinking that we should `rm -fr $path/*` to nuke the contents but leave the directory itself alone.

Comment (mjmac) [reply src/tests/ftest/util/server_utils.py]: Thought about it some more and decided that you're right. Best to unmount/wipe first if there's a device, then just nuke the directory. I was thinking that we might want to preserve the directory but nuke the contents, but I think ultimately we'll want to ensure that the directory gets recreated by the control plane.

**Review by kjacque — APPROVED** [review]

File: src/tests/ftest/util/server_utils.py [hunk 2]
```diff
@@ -230,6 +230,20 @@ def clean_files(self, verbose=True):
             verbose (bool, optional): display clean commands. Defaults to True.
         """
         clean_commands = []
+
+        if self.manager.job.yaml.control_metadata.device:
```
Comment (daltonbohning) [inline src/tests/ftest/util/server_utils.py:234]: Nothing in ftest is yet using `device`, right?

Comment (mjmac) [reply src/tests/ftest/util/server_utils.py:234]: Right. This is for the vos_on_blob branch, where we need at least the external directory for control metadata to be working soon. The plan is to eventually start testing with the control metadata device as well.

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**